hermes-common/src/main/java/pl/allegro/tech/hermes/common/metric/HermesMetrics.java

```diff
@@ -153,7 +158,12 @@ public double getBufferAvailablesBytes() {
     }
 
     private double getDoubleValue(String gauge) {
-        return (double) metricRegistry.getGauges().get(pathCompiler.compile(gauge)).getValue();
+        try {
+            return (double) metricRegistry.getGauges().get(pathCompiler.compile(gauge)).getValue();
+        } catch (NullPointerException exception) {
+            logger.warn("Specified key is null for gauge: {}", gauge);
```

Comment:
1. Otherwise hermes-benchmark will fail at shutdown because of `java.lang.NullPointerException` threw at this point.
2. When we call `getGauges()` we get `String<String gaugeName, Gauge gauge>` which is a map, when we call `get()` on element in map that doesn't exist we will get `null`, then `getValue()` called on null will cause NullPointerException.
3. That can be misleading, changed message to `Specified gauge doesn't exist: {}`

Reply:
maybe it would be better to check if gauge is null instead of catching an exception?

Reply:
OK, I understand now. I suspect that the problem is when we try to call `pl.allegro.tech.hermes.frontend.server.HermesShutdownHandler#isBufferEmpty`. If that's the case what do you think about setting `FRONTEND_GRACEFUL_SHUTDOWN_ENABLED` to false in benchmarks?

Reply:
Good idea, done